sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -0,0 +1,328 @@
+  }
+
+  def literalOfTypeUnsupportedError(valueType: String, ctx: TypeConstructorContext): Throwable = {
+    new ParseException(s"Literals of type '$valueType' are currently not supported.", ctx)
+  }
+
+  def parsingValueTypeError(
+      e: IllegalArgumentException, valueType: String, ctx: TypeConstructorContext): Throwable = {
+    val message = Option(e.getMessage).getOrElse(s"Exception parsing $valueType")
+    new ParseException(message, ctx)
+  }
+
+  def numericLiteralOutTypeRangeError(rawStrippedQualifier: String, minValue: BigDecimal,
+      maxValue: BigDecimal, typeName: String, ctx: NumberContext): Throwable = {
+    new ParseException(s"Numeric literal $rawStrippedQualifier does not " +
+      s"fit in range [$minValue, $maxValue] for type $typeName", ctx)
+  }
+
+  def moreThanOneFromToUnitInIntervalLiteralError(ctx: ParserRuleContext): Throwable = {
+    new ParseException("Can only have a single from-to unit in the interval literal syntax", ctx)
+  }
+
+  def invalidIntervalLiteralError(ctx: IntervalContext): Throwable = {
+    new ParseException("at least one time unit should be given for interval literal", ctx)
+  }
+
+  def invalidIntervalFormError(value: String, ctx: MultiUnitsIntervalContext): Throwable = {
+    new ParseException("Can only use numbers in the interval value part for" +
+      s" multiple unit value pairs interval form, but got invalid value: $value", ctx)
+  }
+
+  def fromToUnitIsStringError(ctx: IntervalValueContext): Throwable = {
+    new ParseException("The value of from-to unit must be a string", ctx)
+  }
+
+  def fromToIntervalUnsupportedError(
+      from: String, to: String, ctx: UnitToUnitIntervalContext): Throwable = {
+    new ParseException(s"Intervals FROM $from TO $to are not supported.", ctx)
+  }
+
+  def dataTypeUnsupportedError(dataType: String, ctx: PrimitiveDataTypeContext): Throwable = {
+    new ParseException(s"DataType $dataType is not supported.", ctx)
+  }
+
+  def partitionTransformNotExpectedError(
+      name: String, describe: String, ctx: ApplyTransformContext): Throwable = {
+    new ParseException(s"Expected a column reference for transform $name: $describe", ctx)
+  }
+
+  def tooManyArgumentsForTransformError(name: String, ctx: ApplyTransformContext): Throwable = {
+    new ParseException(s"Too many arguments for transform $name", ctx)
+  }
+
+  def notEnoughArgumentsForTransformError(name: String, ctx: ApplyTransformContext): Throwable = {
+    new ParseException(s"Not enough arguments for transform $name", ctx)
+  }
+
+  def invalidBucketsNumberError(describe: String, ctx: ApplyTransformContext): Throwable = {
+    new ParseException(s"Invalid number of buckets: $describe", ctx)
+  }
+
+  def invalidTransformArgumentError(ctx: TransformArgumentContext): Throwable = {
+    new ParseException("Invalid transform argument", ctx)
+  }
+
+  val locationMsg = "please use the LOCATION clause to specify it"
+
+  val ownerMsg = "it will be set to the current user"
```

Comment:
ditto. We can directly specify the error message when calling `cannotCleanReservedNamespacePropertyError`